engine/src/main/java/io/camunda/zeebe/engine/processing/job/JobFailProcessor.java

```diff
@@ -102,6 +102,7 @@ private void failJob(final TypedRecord<JobRecord> record) {
       final String errorMessage = String.format(NO_JOB_FOUND_MESSAGE, jobKey);
       rejectionWriter.appendRejection(record, RejectionType.NOT_FOUND, errorMessage);
       responseWriter.writeRejectionOnCommand(record, RejectionType.NOT_FOUND, errorMessage);
+      return;
```

Comment:
@remcowesterhoud the `JobCommandPreconditionChecker` is only used in the `JobFailProcessor` and `JobYieldProcessor` right now. And from these two processors, we only check authorizations in the `JobFailProcessor`.

Furthermore, we're not using the `JobCommandPreconditionChecker` in other job processors since the error messages are not aligned.

So I would leave this for a later refactoring, where we also bring the `JobCommandPreconditionChecker` to all job-related processors. WDYT?

Reply:
Fine with me

Reply:
Created https://github.com/camunda/zeebe/issues/14524 as a tech debt issue.